hail/python/hail/experimental/db.py

```diff
@@ -1,49 +1,154 @@
+    dict mapping dataset names to configurations. Facilitates creation of
+    Dataset objects with `Dataset.from_name_and_json()` method by providing
+    versions of datasets only if they are available in the specified region via
+    the `get_region()` method.
+    """
     @staticmethod
     def from_json(doc):
+        """Create `DatasetVersion` object from dictionary.
+
+        Parameters
+        ----------
+        doc : :obj:`dict`
+            Dictionary containing url and version keys.
+
+        Returns
+        -------
+        :class:`DatasetVersion`
+        """
         assert 'url' in doc, doc
         assert 'version' in doc, doc
         return DatasetVersion(doc['url'],
                               doc['version'])
 
+    @staticmethod
+    def get_region(name, versions, region):
+        """Get versions of a `Dataset` in the specified region, if they exist.
+
+        Parameters
+        ----------
+        name : :obj:`str`
+            Name of dataset.
+        versions : :obj:`list of DatasetVersion`
+            List of DatasetVersion objects where the value for
+            DatasetVersion.url is a dictionary containing the regions 'us' and 'eu'.
+        region : :obj: `str`
+            GCP region from which to access data, available regions given in
+            hl.experimental.DB._valid_regions, currently either 'us' or 'eu'.
+
+        Returns
+        -------
+        available_versions : :obj:`list of DatasetVersion`
+            List of available versions of a `Dataset` for region.
+        """
+        available_versions = []
+        if region is not None:
+            for version in versions:
+                if version.in_region(name, region):
+                    version.url = version.url[region]
+                    available_versions.append(version)
+        return available_versions
+
     def __init__(self, url, version):
         self.url = url
         self.version = version
 
+    def in_region(self, name, region):
+        """To check if a `DatasetVersion` object is accessible in the desired
+        region.
+
+        Parameters
+        ----------
+        name : :obj:`str`
+            Name of dataset.
+        region : :obj: `str`
+            GCP region from which to access data, available regions given in
+            hl.experimental.DB._valid_regions, currently either 'us' or 'eu'.
+
+        Returns
+        -------
+        valid_region : :obj:`bool`
+            Whether or not the dataset exists in the specified region.
+        """
+        current_version = self.version
+        available_regions = [k for k, v in self.url.items() if v is not None]
+        valid_region = region in available_regions
+        if not valid_region:
+            message = f'dataset: \'{name}\', version: {current_version}, exists but is not yet ' \
+                      f'available in \'{region}\' region bucket, ' \
+                      f'currently available in {available_regions} region bucket(s).'
```

Comment:
Is there a specific method of contact to encourage? Otherwise I'll just include they should reach out at https://discuss.hail.is/ or https://hail.zulipchat.com/ to request it.

Reply:
Modified the message, and included a link to the forum in it.